core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -300,10 +300,7 @@ public DeleteWriteBuilder equalityFieldIds(int... fieldIds) {
 
       meta("delete-type", "position");
 
-      if (rowSchema != null) {
-        Preconditions.checkState(createWriterFunc != null,
-            "Cannot create delete file with deletes rows unless createWriterFunc is set");
-
+      if (rowSchema != null && createWriterFunc != null) {
```

Comment:
Why change this ? IMO, if `rowSchema` is not null and `createWriteFunc` is null, we should throw exception, rather than going to the `delete files without row` path ?

Reply:
Using `forTable` sets the row schema automatically, so we can't determine that the user intended to write rows that way.